client/lib/cart-values/cart-items.js

```diff
@@ -412,6 +412,10 @@ export function hasDomainRegistration( cart ) {
 	return some( getAllCartItems( cart ), isDomainRegistration );
 }
 
+export function hasDomainTransfer( cart ) {
```

Comment:
We already have `hasTransferProduct`. While this name seems to be more consistent with other domain-related checks here, I'd prefer to avoid duplicating code. At the very least, make the new function use the old one ;)